### DISCUSSION TITLE
Error in make check

### DISCUSSION THREAD

**Discussion by We-xinyu** [esmf-org/esmf#250] [Building / Installing] [answered]
### Requirements

- [X] Reviewed [ESMF User's Guide](https://earthsystemmodeling.org/doc/)
- [X] Searched [GitHub Discussions](https://github.com/orgs/esmf-org/discussions?discussions_q=)

### ESMF Version

7_0_2

### Issue

When compiling ESMF, make has succeeded, but make check appears as
/usr/bin/ld: cannot find -lnetcdf_c++
collect2: error: ld returned 1 exit status
What should I do to change this error?
![Uploading 1716719851780.jpg…]()

### Autotag

@oehmke

**Comment by danrosen25** [comment]
Hi @We-xinyu,
It doesn't look like the attached image uploaded correctly. In this case the linker, "LD - GNU Linker", cannot find a library named `libnetcdf_c++...` during the make check phase. The easiest way to provide NetCDF paths to the ESMF build is to set `ESMF_NETCDF=nc-config` in your environment. This requires that your environment PATH includes the path to `nc-config`. Is this how you're setting the paths to NetCDF in your build?
[https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#SECTION000104200000000000000](https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#SECTION000104200000000000000)

**Reply by We-xinyu** [reply]
Hi@danrosen25 
Thanks for your reply!
I changed the environment variable of NETCDF_PATH to NETCDF_PATH="split", after that I tried make check then it shows that most of the tests pass, but there are still more than 100 tests that don't pass, does this affect the subsequent use?

**Reply by danrosen25** [reply]
@We-xinyu 
Depending on the platform (system + compiler + libraries) we do see some test failures. We support many configurations that have clean tests, [here are the results from the latest release](https://earthsystemmodeling.org/release/platforms_8_6_1). ESMF v7.0.2 was released in 2017 so it's pretty outdated. I would recommend upgrading to v8.6.1.

**Reply by We-xinyu** [reply]
@danrosen25 Thank you very much for your answer！
I am trying to make map file of CLM model so I use ESMF, but at present there is still a problem in the environment variable setting of ESMF, which leads me not to generate corresponding map file in the end, I don't know where is the problem, this is my ESMF environment configuration, please take a look to see if there is any problem in the environment configuration?
![esmf](https://github.com/esmf-org/esmf/assets/142372239/c415d92c-de6b-42c5-b09e-bda9605963ea)

**Reply by danrosen25** [reply, marked as answer]
@We-xinyu 
Use the defaults as much as possible unless specifically needed. The ESMF build system will scan your system and pick the best configuration. Build option `g` is intended to test the ESMF framework itself so you don't need this unless you're trying to debug ESMF. 
```
export ESMF_DIR="/home/u4/app_esmf/esmf-ESMF_7_0_2"
export ESMF_BOPT="O" # or g
export ESMF_COMM="mpiuni"
export ESMF_COMPILER="gfortran"
export ESMF_NETCDF="nc-config" # nc-config must be part of your path

export ESMF_INSTALL_MODDIR="mod"
export ESMF_INSTALL_BINDIR="bin"
export ESMF_INSTALL_LIBDIR="lib"
export ESMF_INSTALL_PREFIX=<THE ROOT DIRECTORY FOR YOUR ESMF INSTALLATION>
```

**Reply by danrosen25** [reply]
@We-xinyu 
Let me know if the revised build options along with an updated ESMF version (preferably v8.6.1) resolve the issues you're encountering. If you're seeing another issue related to remapping/regridding then we should open a new discussion within that category.

**Reply by We-xinyu** [reply]
Dear @danrosen25 
I reinstalled ESMF with a new version of ESMF 8.6.0, but there are still two places where make check shows failure, but I can't find out where exactly this fails, so I tried to install it and eventually succeeded, I'll put a screenshot of my ESMF environment variables and the places where make check went wrong below, you can see it.
Besides that, I tried to make CLM ground data but it failed eventually, I don't know if it's a problem with ESMF, so I also put the screenshot of the error message below, you help to look at it, thank you very much!
![1717061652435](https://github.com/esmf-org/esmf/assets/142372239/8c51a3b5-8e5e-4477-a530-b1e9ac77818c)
![1717061704299](https://github.com/esmf-org/esmf/assets/142372239/2da8fbb0-3723-4b1d-b8c5-ef20dfd0a1b4)
![1717061724413](https://github.com/esmf-org/esmf/assets/142372239/4c1786c3-4ca3-4bac-b9e0-cdd1b9c7f50f)
![1717061744689](https://github.com/esmf-org/esmf/assets/142372239/723bf021-d5b7-42d3-a46e-17e477c7668e)

**Comment by danrosen25** [comment]
@We-xinyu 
Based on these issues the build looks fine and the regridding failed, likely due to the characteristics of the mesh in the input file. I'm going to close this discussion as resolved but can you open a new discussion under the `regridding` category for `degenerate element detected` and post it the third and fourth images to that discussion? Someone from our team will help explain this error in more detail once you open the new discussion.
